### PR TITLE
Integrate slack notifications

### PR DESCRIPTION
**Pull request by n-dusan**
- Enable slack notifications for errors during upgrade package. To enable slack notifications, pass down a --slack-webhook-url argument or set a SLACK_WEBHOOK_URL environment variable. 
- Remove Python 3.6 from test suite, as ubuntu-latest does not support 3.6.
- Updated testing wheels to support Python 3.10
- Added DEV and PROD environments in the notification message based on Cloudsmith url, so we are able to see where the error is.

## Demo of upgrade script notification during local upgrade

- [ ] set SLACK_WEBHOOK_URL
- [ ] run `python -m upgrade.scripts.upgrade_python_package taf==0.15.0 --update-all --log-location ../upgrade.log`
- [ ] verify that error occurred during upgrade and slack notification is sent.

image

Error in `upgrade.log`
```

  [12 lines of output]
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "...\Local\Temp\pip-install-xp7kr1w0\taf_534d2b82693e49c1855301d66593217f\setup.py", line 105, in <module>
      tests_exist = find_spec("taf.tests")
    File "...\Local\Programs\Python\Python310\lib\importlib\util.py", line 94, in find_spec
      parent = __import__(parent_name, fromlist=['__path__'])
    File "...\Local\Temp\pip-install-xp7kr1w0\taf_534d2b82693e49c1855301d66593217f\taf\__init__.py", line 73, in <module>
      _tuf_patches()
    File "...\Local\Temp\pip-install-xp7kr1w0\taf_534d2b82693e49c1855301d66593217f\taf\__init__.py", line 69, in _tuf_patches
      tuf.repository_lib.get_metadata_fileinfo
  AttributeError: module 'tuf.repository_lib' has no attribute 'get_metadata_fileinfo'. Did you mean: 'get_metadata_versioninfo'?
  [end of output]
```

### CODE REVIEW

File: upgrade/scripts/upgrade_python_package.py [hunk 1]
```diff
@@ -530,6 +580,11 @@ def try_running_module(wheel, *args):
     action="store_true",
     help="Indicates that all packages should be updated",
 )
+parser.add_argument(
+    "--slack-webhook-url",
+    action="store_true",
+    help="Indicates that all failed updates should be sent to slack",
```
Comment (renatav) [inline upgrade/scripts/upgrade_python_package.py]: Very minor - this url is not just a flag that indicates if we should send a notification or not, right?

**Review by renatav — APPROVED** [review]
Nie work! You tested on vagrant? One very minor doc update, but approving

File: upgrade/scripts/upgrade_python_package.py (hunk 1 above)
Comment (n-dusan) [reply upgrade/scripts/upgrade_python_package.py]: Notifications are currently sent If a user provides `--slack-webhook-url` or if SLACK_WEBHOOK_URL environment variable is set. Do you think we maybe need another flag for determining if script should send notifications?

Comment (renatav) [reply upgrade/scripts/upgrade_python_package.py]: No, I'm only thinking that this text suggests that this argument is a flag, but it's actually full url, right? and if the url is not provided, the notification won't be sent, which makes sense

**Comment by n-dusan** [comment]
Fixed parser argument and added (ubuntu@X.X.X.X). How does it look @renatav?

**Review by renatav — APPROVED** [review]
We could enhance this by also sending a notification on first successful update following one or more failed ones. In that case, it is important to save the previous execution result. But we can do that later, we can just create an issue

**Pull request merged**